### PR TITLE
ci: add spell checking with `cspell`

### PR DESCRIPTION
**Pull request by adamhamlin**
> [!TIP]
> The owner of this PR can publish a _preview release_ by commenting `/publish` in this PR. Afterwards, anyone can try it out by running `pnpm add radashi@pr<PR_NUMBER>`.

## Summary

Adds `cspell` to lint process to enforce spelling in virtually all files. Fixed any outstanding spelling mistakes/alarms.

## Related issue, if any:

See https://github.com/orgs/radashi-org/discussions/171 for more info/discussion.

## For any code change,

N/A

- [ ] Related documentation has been updated, if needed
- [ ] Related tests have been added or updated, if needed
- [ ] Related benchmarks have been added or updated, if needed

## Does this PR introduce a breaking change?

No

## Bundle impact

| Status | File | Size  [^1337] | Difference (%) |
|---|---|---|---|
| M | `src/async/defer.ts` | 415 | +0 (+0%) |
| M | `src/curry/partob.ts` | 62 | +0 (+0%) |
| M | `src/object/get.ts` | 190 | +0 (+0%) |
| M | `src/string/camel.ts` | 357 | +0 (+0%) |
| M | `src/string/capitalize.ts` | 152 | +0 (+0%) |
| M | `src/string/dash.ts` | 334 | +0 (+0%) |
| M | `src/string/similarity.ts` | 455 | +0 (+0%) |
| M | `src/string/snake.ts` | 441 | +0 (+0%) |

[^1337]: Function size includes the `import` dependencies of the function.

### CODE REVIEW

**Comment by aleclarson** [comment]
This is awesome!

File: README.md [hunk 1]
```diff
@@ -83,13 +85,13 @@ deno add @radashi-org/radashi
 
 ## FAQ
 
-- **“I need XYZ, but Radashi doesn‘t have it.”**
```
Comment (aleclarson) [inline README.md:86]: Did CSpell remove these trailing spaces? They are Markdown sensitive

File: src/object/get.ts [hunk 2]
```diff
@@ -28,11 +28,11 @@ export function get<TDefault = unknown>(
     if (current === undefined) {
       return defaultValue as TDefault
     }
-    const dequoted = key.replace(/['"]/g, '')
-    if (dequoted.trim() === '') {
+    const withoutQuotes = key.replace(/['"]/g, '')
```
Comment (aleclarson) [inline src/object/get.ts:31]: unquotedKey?

File: README.md (hunk 1 above)
Comment (adamhamlin) [reply README.md:86]: No, I have a vscode extension that did it. What do you mean they are markdown sensitive? Want me to revert?

File: src/object/get.ts (hunk 2 above)
Comment (adamhamlin) [reply src/object/get.ts:31]: Sure

File: README.md (hunk 1 above)
Comment (aleclarson) [reply README.md:86]: In Markdown, two trailing spaces ensure a line break. Without them, this line and the next one will be merged at render-time.

Comment (adamhamlin) [reply README.md:86]: Good to know, will revert

Comment (adamhamlin) [reply README.md:86]: Still a change for this line because of a non-standard quote char

**Comment by adamhamlin** [comment]
@aleclarson I'm assuming the issues with the failing checks are unrelated to this PR

**Review by aleclarson — APPROVED** [review]

**Comment by aleclarson** [comment]
Btw, great job on this, @adamhamlin 👏  You nailed it

**Pull request merged**